cmd/buildah/commit.go

```diff
@@ -112,6 +123,36 @@ func commitListFlagSet(cmd *cobra.Command, opts *commitInputOptions) {
 	flags.BoolVarP(&opts.quiet, "quiet", "q", false, "don't output progress information when writing images")
 	flags.StringVar(&opts.referenceTime, "reference-time", "", "set the timestamp on the image to match the named `file`")
 	_ = cmd.RegisterFlagCompletionFunc("reference-time", completion.AutocompleteNone)
+
+	flags.StringVar(&opts.pull, "pull", "true", "pull the image from the registry if newer or not present in store, if false, only pull the image if not present, if always, pull the image even if the named image is present in store, if never, only use the image present in store if available")
+	flags.Lookup("pull").NoOptDefVal = "true" //allow `--pull ` to be set to `true` as expected.
+
+	flags.BoolVar(&opts.pullAlways, "pull-always", false, "pull the image even if the named image is present in store")
+	if err := flags.MarkHidden("pull-always"); err != nil {
+		panic(fmt.Sprintf("error marking the pull-always flag as hidden: %v", err))
+	}
+	flags.BoolVar(&opts.pullNever, "pull-never", false, "do not pull the image, use the image present in store if available")
+	if err := flags.MarkHidden("pull-never"); err != nil {
+		panic(fmt.Sprintf("error marking the pull-never flag as hidden: %v", err))
+	}
+
+	flags.StringVar(&opts.sbomPreset, "sbom", "", "scan working container using `preset` configuration")
+	_ = cmd.RegisterFlagCompletionFunc("sbom", completion.AutocompleteNone)
+	flags.StringVar(&opts.sbomScannerImage, "sbom-scanner-image", "", "scan working container using scanner command from `image`")
+	_ = cmd.RegisterFlagCompletionFunc("sbom-scanner-image", completion.AutocompleteNone)
+	flags.StringArrayVar(&opts.sbomScannerCommand, "sbom-scanner-command", nil, "scan working container using `command` in scanner image")
+	_ = cmd.RegisterFlagCompletionFunc("sbom-scanner-command", completion.AutocompleteNone)
+	flags.StringVar(&opts.sbomMergeStrategy, "sbom-merge-strategy", "", "merge scan results using `strategy`")
+	_ = cmd.RegisterFlagCompletionFunc("sbom-merge-strategy", completion.AutocompleteNone)
+	flags.StringVar(&opts.sbomOutput, "sbom-output", "", "save scan results to `file`")
+	_ = cmd.RegisterFlagCompletionFunc("sbom-output", completion.AutocompleteDefault)
+	flags.StringVar(&opts.sbomImgOutput, "sbom-image-output", "", "add scan results to image as `path`")
+	_ = cmd.RegisterFlagCompletionFunc("sbom-image-output", completion.AutocompleteNone)
+	flags.StringVar(&opts.sbomPurlOutput, "sbom-purl-output", "", "save scan results to `file``")
+	_ = cmd.RegisterFlagCompletionFunc("sbom-purl-output", completion.AutocompleteDefault)
+	flags.StringVar(&opts.sbomImgPurlOutput, "sbom-image-purl-output", "", "add scan results to image as `path`")
+	_ = cmd.RegisterFlagCompletionFunc("sbom-image-purl-output", completion.AutocompleteNone)
+
```

Comment:
Does commit really pull the container image, or just the sbom image?  I think only the sbom image as the container image is being created by the commands.

Reply:
That's right.  Does this read as implying otherwise?

Reply:
Ah, okay.  The main discussion truncated the highlighted area.  Changing it.

Reply:
I thought that was happening, but wasn't sure.  Part of it is the SBOM is a new concept and I got uber focused on why commit would pull a container image.  

In the Build doc, you added:  "base and SBOM scanner images".  What would you think about changing "image" to "SBOM scanner image" here?  At least for the first instance of "image" in the option text.  I'll go add a suggestion above.

Reply:
Just updated the three places where we describe what a "pull" flag does, hopefully to describe the applicable cases and make them all read more closely to each other.